### PR TITLE
Look for bin/binaryen_js.js directly, instead of expecting us to copy it.

### PR DESCRIPTION
**Pull request by kripken**
Without this, just building and running the test suite didn't work.

### CODE REVIEW

File: scripts/test/shared.py [hunk 1]
```diff
@@ -184,7 +184,9 @@ def is_exe(fpath):
 WASM_METADCE = [os.path.join(options.binaryen_bin, 'wasm-metadce')]
 WASM_EMSCRIPTEN_FINALIZE = [os.path.join(options.binaryen_bin,
                                          'wasm-emscripten-finalize')]
-BINARYEN_JS = os.path.join(options.binaryen_root, 'out', 'binaryen.js')
+# Due to cmake limitations, we emit binaryen_js.js (see CMakeLists.txt
+# for why).
+BINARYEN_JS = os.path.join(options.binaryen_root, 'bin', 'binaryen_js.js')
```
Comment (sbc100) [inline scripts/test/shared.py]: options.binaryen_bin like above?  Or does it not get installed?

Comment (kripken) [reply scripts/test/shared.py]: I think it does, thanks, fixing.

**Comment by dcodeIO** [comment]
Makes me wonder whether `file(RENAME ...)`ing the file once compilation is complete would be more straight-forward, so everyone building it just gets the file name they'd expect. I assume there is a problem with that?

**Comment by kripken** [comment]
@dcodeIO If that works it sounds good. I don't know much cmake though, and after reading the docs, I'm still not sure how to make that happen after the build? How does it order the operations?

**Comment by kripken** [comment]
Let's merge this in for now, as it improves the status quo, but if we can do even better with a cmake file rename then let's look into that too!

**Comment by dcodeIO** [comment]
Appears we may instead append

```cmake
set_target_properties(binaryen_js PROPERTIES OUTPUT_NAME "binaryen")
```

to get

```
[ 98%] Linking CXX executable bin/binaryen.js
```

Documentation says

> OUTPUT_NAME sets the real name of a target when it is built and can be used to help create two targets of the same name even though CMake requires unique logical target names.

I don't know much about CMake and whether there may be issues with this, but works for me locally.

**Pull request merged**